### PR TITLE
Add Atlas/v3 auth support

### PR DESCRIPTION
**Pull request by johlo**

### CODE REVIEW

**Comment by LuddeFro** [comment]
lgtm

File: README.md [hunk 1]
```diff
@@ -16,16 +14,20 @@ To compile the client:
...
 
+### With Atlas/v3 API authentication secret 
 
-## Running
 To start the client do:
 
- `$ ./push-api-client --client-id=$CLIENT_ID --client-secret=$CLIENT_SECRET --subscription-file=sample_subscription.json`
+ `$ ./push-api-client --secret=$CLIENT_SECRET --subscription-file=sample_subscription_v3.json`
+
+where `CLIENT_SECRET` is the same that you already use to access v3 of the abios REST API . The `sample_subscription_v3.json` file contains a simple subscription specification that will listen to all events from the `series_updates` channel (for the games your account has access to).
```
Comment (ptorsson) [inline README.md]: `where `CLIENT_SECRET` is the same that you already use to access v3 of the Abios REST API.`

File: http_requests.go [hunk 2]
```diff
@@ -12,35 +13,57 @@ import (
...
-	} else if err != nil {
-		fmt.Printf("%s [ERROR]: Failed to connect to WS url. Error='%s'\n",
-			time.Now().Format(timestampMillisFormat), err.Error())
-		return nil, err
+	conn, resp, err := dialer.Dial(URL, h)
+
+	if err != nil {
+		if resp != nil {
+			return nil, WebsocketSetupHTTPError{HttpStatus: resp.StatusCode, error: err}
+		} else {
```
Comment (ptorsson) [inline http_requests.go:51]: IMO `else` can be removed but it's up to you.

File: README.md (hunk 1 above)
Comment (johlo) [reply README.md]: Fixed

File: http_requests.go [hunk 3]
```diff
@@ -75,21 +99,17 @@ func fetchSubscriptions(accessToken string) ([]byte, error) {
 	return respBody, err
 }
 
-func registerSubscription(accessToken string, sub Subscription) (uuid.UUID, bool, error) {
-	URL := buildHTTPURLFromWSURL(*addrFlag)
-	URL = URL + "/subscription"
-	URL = URL + "?access_token=" + accessToken
-
+func registerSubscription(sub Subscription) (uuid.UUID, bool, error) {
 	j, _ := json.Marshal(sub)
```
Comment (ptorsson) [inline http_requests.go:108]: Missing error check

File: http_requests.go [hunk 4]
```diff
@@ -132,21 +152,18 @@ func registerSubscription(accessToken string, sub Subscription) (uuid.UUID, bool
...
 }
 
-func updateSubscription(accessToken string, sub Subscription) (uuid.UUID, bool, error) {
-	URL := buildHTTPURLFromWSURL(*addrFlag)
-	URL = URL + "/subscription/" + sub.ID.String()
-	URL = URL + "?access_token=" + accessToken
-
+func updateSubscription(sub Subscription) (uuid.UUID, bool, error) {
+	endpoint := "/subscription/" + sub.ID.String()
 	j, _ := json.Marshal(sub)
```
Comment (ptorsson) [inline http_requests.go]: Missing error check

File: http_requests.go (hunk 2 above)
Comment (johlo) [reply http_requests.go:51]: Let's keep it as is, but your comment made me realize I had forgotten to handle the error correctly in the caller. :+1:

File: http_requests.go (hunk 4 above)
Comment (johlo) [reply http_requests.go]: Fixed

**Review by ptorsson — APPROVED** [review]

**Pull request merged**